modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/dataframe.py

```diff
@@ -646,26 +646,19 @@ def generate_by_name(by):
 
         agg_exprs = OrderedDict()
         if isinstance(agg, str):
-            if agg == "nlargest" or agg == "nsmallest":
-                n = kwargs["agg_kwargs"]["n"]
-                if agg == "nsmallest":
-                    n = -n
-                n = LiteralExpr(n)
-                for col in agg_cols:
-                    agg_exprs[col] = AggregateExpr(agg, [base.ref(col), n])
-            else:
-                for col in agg_cols:
-                    agg_exprs[col] = AggregateExpr(agg, base.ref(col))
+            col_to_ref = {col: base.ref(col) for col in agg_cols}
+            self._add_agg_exprs(agg, col_to_ref, kwargs, agg_exprs)
         else:
             assert isinstance(agg, dict), "unsupported aggregate type"
             multiindex = any(isinstance(v, list) for v in agg.values())
-            for k, v in agg.items():
-                if isinstance(v, list):
-                    for item in v:
-                        agg_exprs[(k, item)] = AggregateExpr(item, base.ref(k))
+            for col, aggs in agg.items():
+                if isinstance(aggs, list):
+                    for a in aggs:
+                        col_to_ref = {(col, a): base.ref(col)}
+                        self._add_agg_exprs(a, col_to_ref, kwargs, agg_exprs)
```

Comment:
we only need agg kwargs there
```suggestion
                        self._add_agg_exprs(a, col_to_ref, kwargs["agg_kwargs"], agg_exprs)
```

Reply:
Right, but only for 2 aggregates, for all others it's not required at all.

Reply:
yes, I mean that passing the whole `kwargs` dict is excessive and misleading, as it implies that the function may rely on all the groupby kwargs, when in reality it only need kwargs of the aggregation function

Reply:
Right, I mean, that getting `agg_kwargs` from `kwargs` is only required for 2 aggregates. In all other cases this operation is redundant. I understand that this is a cheep operation, but I prefer to avoid unnecessary computations.

Reply:
ok then